### DISCUSSION TITLE
Creating fewer migrations

### DISCUSSION THREAD

**Discussion by einarpersson** [pocketbase/pocketbase#3963] [Q&A]
Consider that I want to change API-rules for all my collections (for some reason), so I go into the admin interface and click through each. This causes one migration file to be created for every change, but i would like to compress that into one migration rather (since it conceptually is one) before commiting to git.

What do I do?
1. Open the migration folder and merge the files manually?
2. Use some CLI-command that I have not understood how to use

Thanks!

**Comment by ganigeorgiev** [comment]
You can merge them manually OR delete the migrations and run [`./pocketbase migrate collections`](https://pocketbase.io/docs/js-migrations/#collections-snapshot) to create a full snapshot of your latest collections and commit only that.
To cleanup the local migrations history from the deleted migrations, you can run [`./pocketbase migrate history-sync`](https://pocketbase.io/docs/js-migrations/#migrations-history)

The drawback of having a full snapshot migration instead of the more granular ones is that if there are multiple people working on the project and your migrations is the last one it may overwrite any other schema changes that you may not have pulled and reflected in your snapshot before making your merge.
